ui/src/shared/reducers/predicates.test.ts

```diff
@@ -18,51 +13,57 @@ import {
-  it('should set the isSerious property', () => {
+describe('Predicates reducer', () => {
+  it('Can set the isSerious property', () => {
     expect(initialState.isSerious).toEqual(false)
     let result = predicatesReducer(initialState, setIsSerious(true))
     expect(result.isSerious).toEqual(true)
     result = predicatesReducer(initialState, setIsSerious(false))
     expect(result.isSerious).toEqual(false)
   })
-  it('should set the bucketName property', () => {
+
+  it('Can set the bucketName property', () => {
     const bucketName = 'bucket_list'
     expect(initialState.bucketName).toEqual('')
     const result = predicatesReducer(initialState, setBucketName(bucketName))
     expect(result.bucketName).toEqual(bucketName)
   })
-  it('should set the timeRange property', () => {
-    expect(initialState.timeRange).toEqual([recently - HOUR_MS, recently])
-    const result = predicatesReducer(initialState, setTimeRange([1000, 2000]))
-    expect(result.timeRange).toEqual([1000, 2000])
+
+  it('Can set the timeRange property', () => {
+    expect(initialState.timeRange).toBeNull()
+
+    const instantiatedPastHour = convertTimeRangeToCustom(pastHourTimeRange)
+    const result = predicatesReducer(
+      initialState,
+      setTimeRange(instantiatedPastHour)
+    )
+    expect(result.timeRange).toEqual(instantiatedPastHour)
   })
-  it('should set the filter property', () => {
-    const filter: Filter = {key: 'mean', equality: '=', value: '100'}
+
+  it('Can set the filter property', () => {
     expect(initialState.filters).toEqual([])
     const result = predicatesReducer(initialState, setFilter(filter, 0))
     expect(result.filters).toEqual([filter])
   })
-  it('should delete a filter that has been set', () => {
-    const filter: Filter = {key: 'mean', equality: '=', value: '100'}
+
+  it('Can delete a filter that has been set', () => {
     let result = predicatesReducer(initialState, setFilter(filter, 0))
     expect(result.filters).toEqual([filter])
     result = predicatesReducer(initialState, deleteFilter(0))
-    expect(initialState.filters).toEqual([])
+    expect(result.filters).toEqual([])
   })
-  it('should reset the state after a filter DWP has been successfully submitted', () => {
-    const state = Object.assign({}, initialState)
-    const filter: Filter = {key: 'mean', equality: '=', value: '100'}
-    initialState.isSerious = predicatesReducer(
-      initialState,
-      setIsSerious(true)
-    ).isSerious
-    initialState.filters = predicatesReducer(
+
+  it('Can reset the state after a filter DWP has been successfully submitted', () => {
+    const intermediateState = predicatesReducer(
       initialState,
       setFilter(filter, 0)
-    ).filters
-    const result = predicatesReducer(initialState, resetPredicateState())
-    expect(result).toEqual(state)
+    )
+    const result = predicatesReducer(intermediateState, resetPredicateState())
+    expect(result).toEqual(initialState)
```

Comment:
should we make a frozen copy of the initialState to ensure that the data we're comparing is referencing a direct copy of the initialState, and not the updated in-memory initialState that may have been updated by `setFilter`?

Reply:
yeah-- that's a good idea ! thanks :)